Yona/Yona/UserRequestManager.swift

```diff
@@ -75,16 +75,10 @@ class UserRequestManager{
     
     
     func postOpenAppEvent(_ user: Users, onCompletion: @escaping APIResponse) {
-        var openAppEventLinkPath: String?
-        for temp in user.devices {
-            if temp.isRequestingDevice! {
-                openAppEventLinkPath = temp.openAppEventLink
-            }
-        }
-        
+        let openAppEventLinkPath = user.devices.first(where: { $0.isRequestingDevice! }).map {$0.openAppEventLink}
         if let path = openAppEventLinkPath, let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"]  as? String, let appVersionCode = Bundle.main.infoDictionary?["CFBundleVersion"]  as? String {
             let bodyForOpenAppEvent = ["operatingSystem": "IOS", "appVersion": appVersion, "appVersionCode":appVersionCode] as BodyDataDictionary
-            genericUserRequest(httpMethods.post, path: path, userRequestType: userRequestTypes.postUser, body: bodyForOpenAppEvent, onCompletion: { (success, message, code, nil) in
+            genericUserRequest(httpMethods.post, path: path!, userRequestType: userRequestTypes.postUser, body: bodyForOpenAppEvent, onCompletion: { (success, message, code, nil) in
```

Comment:
Just curious: 
You changed this
```
path: path
```

into this:
```
path: path!
```

What is the functional difference of that exclamation mark?

Reply:
Exclamation mark used in Swift to unwrapped the optional values. 
Now the type of `path` is a string, not an optional anymore.